WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -262,6 +262,64 @@ def edit
     render :edit
   end
 
+  def payment_setup
+    @competition = Competition.find(params[:id])
+
+    client = create_oauth_client
+    oauth_params = {
+      scope: 'read_write',
+      state: @competition.id,
+    }
+    @authorise_url = client.auth_code.authorize_url(oauth_params)
+  end
```

Comment:
OK, I didn't think of the param could be "unsafe". But loading the object from the DB is an awkward way of sanitizing it. It also makes a nil pointer possible. And can it really do any damage in the OAuth code?

Anyway, using `Competition.exists?(params[:id])` is probably the right thing here.